File: Soulo/Models/SubscriptionStatus.swift
import Foundation

struct SubscriptionInfo: Codable {
    var status: SubscriptionStatus
    var planType: SubscriptionPlan?
    var originalId: String?
    var currentId: String?
    var expiresAt: TimeInterval?
    var autoRenew: Bool
    var trialStart: TimeInterval?
    var trialEnd: TimeInterval?
    var updatedAt: TimeInterval

    static let `default` = SubscriptionInfo(
        status: .free,
        autoRenew: false,
        updatedAt: Date().timeIntervalSince1970
    )

    var isActive: Bool {
        switch status {
        case .active, .trial: return true
        case .free, .expired, .cancelled: return false
        }
    }
}

enum SubscriptionPlan: String, Codable {
    case monthly = "soulo_monthly"
    case annual = "soulo_annual"
    case family = "soulo_family"

    var displayName: String {
        switch self {
        case .monthly: return "Monthly"
        case .annual: return "Annual"
        case .family: return "Family"
        }
    }

    var price: String {
        switch self {
        case .monthly: return "$9.99/month"
        case .annual: return "$79.99/year"
        case .family: return "$14.99/month"
        }
    }

    var monthlyPrice: Double {
        switch self {
        case .monthly: return 9.99
        case .annual: return 6.67
        case .family: return 14.99
        }
    }
}

File: android/app/src/main/java/com/soulo/app/models/SubscriptionStatus.kt
package com.soulo.app.models

import kotlinx.serialization.Serializable

@Serializable
enum class SubscriptionPlan(val storeId: String, val price: Double) {
    monthly("soulo_monthly", 9.99),
    annual("soulo_annual", 79.99),
    family("soulo_family", 14.99);

    val displayName: String get() = when (this) {
        monthly -> "Monthly"
        annual -> "Annual"
        family -> "Family"
    }

    val priceDisplay: String get() = when (this) {
        monthly -> "$9.99/month"
        annual -> "$79.99/year"
        family -> "$14.99/month"
    }
}

@Serializable
data class SubscriptionStatus(
    val isActive: Boolean = false,
    val plan: SubscriptionPlan? = null,
    val expiryDate: Long? = null,
    val entryCount: Int = 0,
    val trialLimit: Int = 7,
    val isFamilyShared: Boolean = false
) {
    val canRecord: Boolean get() = isActive || entryCount < trialLimit
    val remainingTrial: Int get() = (trialLimit - entryCount).coerceAtLeast(0)
}
